File: palindromicSubstrings.cpp
class Solution {
private:
    string str;
    int ans;
public:
    void parindrome_check(int i, int j) {
        if(i < 0 || j >= str.size()) return;
        if(str[i] == str[j]) {
            ans++;
        } else {
            return;
        }
        parindrome_check(--i, ++j);
    }
    int countSubstrings(string s) {
        str = s;
        ans = 0;
        for (int i = 0; i < s.size(); ++i)
        {
            parindrome_check(i, i);
            parindrome_check(i, i+1);
        }
        return ans;
    }
};

File: palindromicSubstrings.java
public class Solution {
    int count = 0;
    String str;
    private void palindromeCheck(int i, int j) {
        if(i < 0 || j >= str.length()) return;
        if(str.charAt(i) == str.charAt(j) ) {
            count++;
        } else {
            return;
        }
        palindromeCheck(--i, ++j);
    }
    
    public int countSubstrings(String s) {
        str = s;
        for(int i = 0; i < str.length(); i++) {
            palindromeCheck(i, i);
            palindromeCheck(i, i+1);
        }
        return count;
    }
}
